java/compaction/compaction-job-execution/src/main/java/sleeper/compaction/job/execution/DefaultSelector.java

```diff
@@ -55,33 +55,30 @@ public DefaultSelector(
     public CompactionRunner chooseCompactor(CompactionJob job) {
         TableProperties tableProperties = tablePropertiesProvider
                 .getById(job.getTableId());
-        String method = tableProperties.get(TableProperty.COMPACTION_METHOD).toUpperCase(Locale.UK);
-
-        // Convert to enum value and default to Java
-        CompactionMethod desired;
-        try {
-            desired = CompactionMethod.valueOf(method);
-        } catch (IllegalArgumentException e) {
-            desired = CompactionMethod.DEFAULT;
-        }
-
-        CompactionRunner defaultRunner = new StandardCompactor(tablePropertiesProvider, stateStoreProvider, objectFactory, configuration);
-        CompactionRunner runner = defaultRunner;
-        switch (desired) {
-            case RUST:
-                runner = new RustCompaction(tablePropertiesProvider, stateStoreProvider);
-                break;
-            default:
-                break;
-        }
+        CompactionMethod method = tableProperties.getEnumValue(COMPACTION_METHOD, CompactionMethod.class);
+        CompactionRunner runner = createRunnerForMethod(method);
 
         // Is an iterator specifed? If so can we support this?
         if (job.getIteratorClassName() != null && !runner.supportsIterators()) {
             LOGGER.debug("Table has an iterator set, which compactor {} doesn't support, falling back to default", runner.getClass().getSimpleName());
-            runner = defaultRunner;
+            runner = createJavaRunner();
         }
 
         LOGGER.info("Selecting {} compactor (language {}) for job ID {} table ID {}", runner.getClass().getSimpleName(), runner.implementationLanguage(), job.getId(), job.getTableId());
         return runner;
     }
+
+    private CompactionRunner createRunnerForMethod(CompactionMethod method) {
+        switch (method) {
+            case RUST:
+                return new RustCompaction(tablePropertiesProvider, stateStoreProvider);
```

Comment:
Is there any scenario where we'd need to create the RustCompaction outside this method in the future, if so moving it out to a separate private method like the java runner might be worth it

Reply:
I think that'd be easy enough that there's no reason to do it now, unless it'd actually read better?

Reply:
Was just a note for future ideas, approved as it was fine without